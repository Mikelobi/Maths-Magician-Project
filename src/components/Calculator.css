.calculator {
  background-color: #111;
  box-shadow: 3px 3px 5px rgba(0, 0, 0, 0.5);
  display: flex;
  flex-wrap: wrap;
  height: 500px;
  width: 290px;
  padding: 5px;
}

button {
  border: none;
  border-radius: 50%;
  cursor: pointer;
  font-size: 24px;
  margin: 5px;
  height: 60px;
  width: 60px;
}

button:active {
  transform: scale(0.93);
}

button:focus {
  outline: none;
}

button:hover {
  opacity: 0.85;
}

button.ac {
  margin-right: 50%;
}

button.light-grey {
  background-color: #9f9f9f;
}

button.orange {
  background-color: #f19a35;
  color: #fff;
}

button.dark-grey {
  background-color: #313131;
  color: #fff;
}

button.double-height {
  border-radius: 40px;
  height: 110px;
}

button.double-width {
  border-radius: 40px;
  width: 130px;
}

.display {
  color: #000;
  font-size: 30px;
  font-weight: bold;
  text-align: right;
  padding: 15px;
  width: 100%;
}

.numbers-container {
  width: 75%;
}

.operations-container {
  width: 25%;
}
