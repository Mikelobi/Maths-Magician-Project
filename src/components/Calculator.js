/* eslint-disable react/destructuring-assignment */

import React from 'react';
import PropTypes from 'prop-types'; /* eslint-disable react/forbid-prop-types */
import calculate from '../logic/calculate';

import './Calculator.css';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      next: '',
      operation: '',
    };
    this.handleClick = this.handleClick.bind(this);
  }

  componentDidUpdate() {
    const { total, next } = this.state;
    const display = document.querySelector('.display');
    if (next != null) {
      // @ts-ignore
      display.value = next;
    } else if (total != null) {
      // @ts-ignore
      display.value = total;
    }
  }

  handleClick = (e) => {
    if (e.target.textContent === 'AC') {
      const display = document.querySelector('.display');
      // @ts-ignore
      display.value = 0;
    }
    const { total, next, operation } = this.state;
    this.setState(calculate({ total, next, operation }, e.target.textContent));
  };

  render() {
    return (
      <div className="calculator">
        <div id="display-result" className="display-result">
          <input className="display" placeholder="0" />
        </div>

        <div className="numbers-container">
          <button
            type="button"
            className="ac double-width light-grey"
            onClick={this.handleClick}
          >
            AC
          </button>
          {this.props.nums.map((num) => (
            <button
              type="button"
              className={`dark-grey ${num === 0 && 'double-width'}`}
              key={num}
              onClick={this.handleClick}
            >
              {num}
            </button>
          ))}
          <button
            type="button"
            className="light-grey"
            onClick={this.handleClick}
          >
            .
          </button>
        </div>
        <div className="operations-container">
          {this.props.ops.map((op) => (
            <button
              type="button"
              className="orange"
              key={op}
              onClick={this.handleClick}
            >
              {op}
            </button>
          ))}
        </div>
      </div>
    );
  }
}
Calculator.propTypes = {
  nums: PropTypes.array.isRequired,
  ops: PropTypes.array.isRequired,
};

export default Calculator;
