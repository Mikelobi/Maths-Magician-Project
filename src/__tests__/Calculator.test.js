import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

it('Renders component correctly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Calculator />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('perform calculator operations', () => {
  test('should display clicked number on calculator display', () => {
    render(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('2'));
    // @ts-ignore
    expect(screen.getByRole('textbox').value).toBe('2');
  });
});
