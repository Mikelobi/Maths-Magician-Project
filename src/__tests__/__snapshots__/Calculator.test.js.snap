// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component correctly 1`] = `
Array [
  <div
    className="header"
  >
    <div>
      <h1
        className="header-title"
      >
        Math Magicians
      </h1>
    </div>
    <nav
      className="navigation"
    >
      <li
        className="navLink"
      >
        <a
          aria-current="page"
          className="link active"
          data-testid={1}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          className="link"
          data-testid={2}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          className="link"
          data-testid={3}
          href="/quotes"
          onClick={[Function]}
        >
          Quotes
        </a>
      </li>
    </nav>
  </div>,
  <div
    className="calc-content"
  >
    <h3>
      Let us do some
       
      <span>
         Math Magic 
      </span>
      !
    </h3>
    <p>
      If you notice any bug while perfoming your
       
      <span>
         Math Magic 
      </span>
      , please raise an issue on the project Github page at this
       
      <a
        href="https://github.com/Mikelobi/Maths-Magician-Project/issues"
      >
        Issue Page
      </a>
       
    </p>
  </div>,
  <div
    className="calculator"
  >
    <div
      className="display-result"
      data-testid="display-result"
    >
      <input
        className="display"
        placeholder="0"
      />
    </div>
    <div
      className="numbers-container"
    >
      <button
        className="double-width light-grey"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="light-grey"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="dark-grey false"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="dark-grey double-width"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="light-grey"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="operations-container"
    >
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="orange"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
